### ISSUE TITLE
Can the convolution module handle heterogeneous graphs? 

### ISSUE THREAD

**Issue by chaosles** [xincoder/GRIP#18]
Can ST-GCN be used if the influence of fixed obstacles is considered when constructing the graph? I want to add the influence of fixed obstacles to predict vehicle trajectory as my research content. Do you have any good suggestions? thank you！

**Comment by xincoder** [comment]
Hi @chaosles , considering fixed obstacles is a good idea. If you want to use the ST-GCN-like structure, you may want to modify the feature representation in our work. Specifically, in our work, all valid (observed) objects are organized in the first few rows in the input (and the following feature maps). In this case, if you want to use ST-GCN convolutional module, you can mark the fixed obstacles (using a specific value, or other ways) and also add one matrix to remember the mapping relation between the ST-GCN locations (2D) and the feature map (1D, each row corresponds to one object).

**Comment by chaosles** [comment]
@xincoder  Thank you for your suggestion. I manually created the location information of the fixed obstacle and added it to the training set.
In addition, I would like to ask whether the experimental results in the paper are the output of the verification set or the output of the test set.

**Comment by xincoder** [comment]
@chaosles Glad to hear that you figured out a way to add obstacles. The results reported in our papers are results on the testing sets.

**Comment by chaosles** [comment]
@xincoder

**Comment by guoyage** [comment]
> @chaosles Glad to hear that you figured out a way to add obstacles. The results reported in our papers are results on the testing sets.
@chaosles@xincoder  I'm very curious about how to obtain the corresponding semantic map or obstacle location information for Apollospace, because I found that the trajectory prediction module on the Apollospace official website only provides the trajectory information of the traffic agent, and does not provide fixed obstacles or  semantic map information, So  I hope to get some help from you, I will be very grateful. Thanks
